### DISCUSSION TITLE
Firebase Storage download file content

### DISCUSSION THREAD

**Discussion by sertiano93** [invertase/react-native-firebase#5092] [Q&A]
I wanted to ask if there is any way to actually download the content of the firebase storage url. 
I am for the moment uploading images to firebase storage, and the recipient is receiving the url which is then connected directly to React Natives Image function. However, I am trying to encrypt the file before uploading it so I would need to actually download the content from the url and decrypt it on client side. Any advices or thoughts ?

**Comment by mikehardy** [comment]
You may like to look out our e2e storage tests which exercise uploading and downloading, they may serve as interesting examples and might hit your use case? 

https://github.com/invertase/react-native-firebase/blob/a61d0cdebb6c2c7c9c3966f0a7a0af6a0ef3ae05/packages/storage/e2e/StorageTask.e2e.js#L43-L51

**Reply by sertiano93** [reply]
Thanks I will try that.
